src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -197,9 +210,24 @@ internal void AddProvider(string workerDir)
             }
         }
 
-        private List<WorkerDescriptionProfile> ReadWorkerDescriptionProfiles(JToken profilesJToken)
+        private static JsonElement GetWorkerConfigJsonElement(string workerConfigPath)
         {
-            var profiles = profilesJToken.ToObject<IList<WorkerProfileDescriptor>>();
+            ReadOnlySpan<byte> jsonSpan = File.ReadAllBytes(workerConfigPath).AsSpan();
+
+            if (jsonSpan.StartsWith(stackalloc byte[] { 0xEF, 0xBB, 0xBF }))
```

Comment:
Replace this with a collection expression, as that will be clean and avoid the allocation:
``` csharp
if (jsonSpan.StartsWith<byte>([0xEF, 0xBB, 0xBF]))
```